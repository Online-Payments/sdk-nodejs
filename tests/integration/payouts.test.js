const sdkInstance = require('./__fixture__');
const config = require('./__fixture__/config');
const stubs = require('./__stubs__');

describe('Payouts', () => {
  describe('Promises', () => {
    test('it can create a payout', async () => {
      const sdkResponse = await sdkInstance.payouts.createPayout(config.merchantId, stubs.createPayout, {});
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });

    test('it can get a payout', async () => {
      const sdkResponse = await sdkInstance.payouts.getPayout(config.merchantId, 'PAYOUT_ID', {});
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });
  });

  describe('Callback', () => {
    test('it can create a payout', (done) => {
      sdkInstance.payouts.createPayout(config.merchantId, stubs.createPayout, {}, (error, sdkResponse) => {
        expect(error).toBeNull();
        expect(sdkResponse.isSuccess).toBe(true);
        expect(sdkResponse).toMatchSnapshot();
        done();
      });
    });

    test('it can get a payout', (done) => {
      sdkInstance.payouts.getPayout(config.merchantId, 'PAYOUT_ID', {}, (error, sdkResponse) => {
        expect(error).toBeNull();
        expect(sdkResponse.isSuccess).toBe(true);
        expect(sdkResponse).toMatchSnapshot();
        done();
      });
    });
  });
});
