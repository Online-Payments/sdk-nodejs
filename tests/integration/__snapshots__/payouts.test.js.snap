// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payouts Callback it can create a payout 1`] = `
Object {
  "body": Object {
    "id": "string",
    "payoutOutput": Object {
      "amountOfMoney": Object {
        "amount": 1000,
        "currencyCode": "EUR",
      },
    },
    "status": "CREATED",
    "statusOutput": Object {
      "isCancellable": true,
      "statusCategory": "CREATED",
      "statusCode": -2147483648,
    },
  },
  "isSuccess": true,
  "status": 201,
}
`;

exports[`Payouts Callback it can get a payout 1`] = `
Object {
  "body": Object {
    "id": "string",
    "payoutOutput": Object {
      "amountOfMoney": Object {
        "amount": 1000,
        "currencyCode": "EUR",
      },
    },
    "status": "CREATED",
    "statusOutput": Object {
      "isCancellable": true,
      "statusCategory": "CREATED",
      "statusCode": -2147483648,
    },
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Payouts Promises it can create a payout 1`] = `
Object {
  "body": Object {
    "id": "string",
    "payoutOutput": Object {
      "amountOfMoney": Object {
        "amount": 1000,
        "currencyCode": "EUR",
      },
    },
    "status": "CREATED",
    "statusOutput": Object {
      "isCancellable": true,
      "statusCategory": "CREATED",
      "statusCode": -2147483648,
    },
  },
  "isSuccess": true,
  "status": 201,
}
`;

exports[`Payouts Promises it can get a payout 1`] = `
Object {
  "body": Object {
    "id": "string",
    "payoutOutput": Object {
      "amountOfMoney": Object {
        "amount": 1000,
        "currencyCode": "EUR",
      },
    },
    "status": "CREATED",
    "statusOutput": Object {
      "isCancellable": true,
      "statusCategory": "CREATED",
      "statusCode": -2147483648,
    },
  },
  "isSuccess": true,
  "status": 200,
}
`;
